lang/c++/api/LogicalType.hh

```diff
@@ -47,17 +47,17 @@ public:
     // Precision must be positive and scale must be either positive or zero. The
     // setters will throw an exception if they are called on any type other
     // than DECIMAL.
-    void setPrecision(int precision);
-    int precision() const { return precision_; }
-    void setScale(int scale);
-    int scale() const { return scale_; }
+    void setPrecision(int64_t precision);
```

Comment:
I think for the scale and precision a int32 is more than enough. Precision probably won't go over 38 and scale not over 77 (or -77).

Reply:
Fair enough, I've instead added `static_cast` to where we read this field